CommunicationLayer/MessagingFramework/BatteryMessage.cpp

```diff
@@ -0,0 +1,246 @@
+}
+
+unsigned short BatteryMessage::lowestCellTemperature() const
+{
+    return getUnsignedShort(messageData_, LOWEST_CELL_TEMPERATURE);
+}
+
+unsigned char BatteryMessage::lowestCellTemperatureNumber() const
+{
+    return getUnsignedChar(messageData_, LOWEST_CELL_TEMPERATURE_NUMBER);
+}
+
+unsigned short BatteryMessage::highestCellTemperature() const
+{
+    return getUnsignedShort(messageData_, HIGHEST_CELL_TEMPERATURE);
+}
+
+unsigned char BatteryMessage::highestCellTemperatureNumber() const
+{
+    return getUnsignedChar(messageData_, HIGHEST_CELL_TEMPERATURE_NUMBER);
+}
+
+unsigned int BatteryMessage::voltage() const
+{
+    return getUnsignedInt(messageData_, VOLTAGE);
+}
+
+unsigned int BatteryMessage::current() const
+{
+    return getUnsignedInt(messageData_, CURRENT);
+}
+
+unsigned short BatteryMessage::fan0Speed() const
+{
+    return getUnsignedShort(messageData_, FAN_0_SPEED);
+}
+
+unsigned short BatteryMessage::fan1Speed() const
+{
+    return getUnsignedShort(messageData_, FAN_1_SPEED);
+}
+
+unsigned short BatteryMessage::fanContactors12VCurrentConsumption() const
+{
+    return getUnsignedShort(messageData_, FAN_AND_CONTACTORS_12V_CURRENT_CONSUMPTION);
+}
+
+unsigned short BatteryMessage::cmu12VCurrentConsumption() const
+{
+    return getUnsignedShort(messageData_, CMU_12V_CURRENT_CONSUMPTION);
+}
+
+bool BatteryMessage::bmsCanLockedOut() const
+{
+    return static_cast<bool>(messageData_.at(BMS_CAN_LOCKED_OUT));
+}
+
+
+QString BatteryMessage::toString() const
+{
+    // TODO chars don't need QString::number
+    QString messageString;
+    messageString += QString::number(MessageDefines::Battery) + ", ";
+    messageString += QString::number(alive()) + ", ";
+    messageString += QString::number(packSocAmpHours()) + ", ";
+    messageString += QString::number(packSocPercentage()) + ", ";
+    messageString += QString::number(packBalanceSoc()) + ", ";
+    messageString += QString::number(packBalanceSocPercentage()) + ", ";
+    messageString += QString::number(chargingCellVoltageError()) + ", ";
+    messageString += QString::number(cellTemperatureMargin()) + ", ";
+    messageString += QString::number(dischargingCellVoltageError()) + ", ";
+    messageString += QString::number(totalPackCapacity()) + ", ";
+    messageString += QString::number(prechargeContactorDriverStatus()) + ", ";
+    messageString += QString::number(prechargeState()) + ", ";
+    messageString += QString::number(prechargeTimerElapsed()) + ", ";
+    messageString += QString::number(prechargeTimerCount()) + ", ";
+    messageString += QString::number(lowestCellVoltage()) + ", ";
+    messageString += QString::number(lowestCellVoltageNumber()) + ", ";
+    messageString += QString::number(highestCellVoltage()) + ", ";
+    messageString += QString::number(highestCellVoltageNumber()) + ", ";
+    messageString += QString::number(lowestCellTemperature()) + ", ";
+    messageString += QString::number(lowestCellTemperatureNumber()) + ", ";
```

Comment:
No, we don't want it represented as a char (this will be in ascii). We want it as a number. Otherwise, 0,1,2,3, is going to show up as their ascii representations.